runtime/validators.go

```diff
@@ -0,0 +1,16 @@
+package runtime
+
+import (
+	"context"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// ValidatorUpdateService is a type injected by the runtime module that can be used by modules to send validator
+// updates back to tendermint.
+type ValidatorUpdateService interface {
+
+	// SetValidatorUpdates sends the provided validator updates back to tendermint. It should only be called during
+	// an end blocker. The current behavior makes it an error for SetValidatorUpdates to be called twice in one block.
+	SetValidatorUpdates(context.Context, []abci.ValidatorUpdate)
```

Comment:
@alexanderbez @marbar3778 this is what I propose instead of having `EndBlocker` in `Handler` have anything to do with tendermint.

Basically the staking module would call `SetValidatorUpdates` in its end blocker rather than having this be a return type from `EndBlocker`.

It may seem like a small issue, but we know there are changes coming in tendermint and in particular `ValidatorUpdate` depends on a specific `PublicKey` struct with a specific list of key types. It's just more future proof to do something like this to provide a stable API for most modules that don't deal with staking.

Reply:
This requires modules to rely on tendermint. I think if we can have a wrapper type or a way for the runtime to call toTmType may abstract this reliance of tendermint from the modules.

Reply:
Makes sense. For now I'm trying to start with the lowest hanging fruit that's generalizable

Reply:
discussed in slack, lets merge and see how things play out and if we still want to abstract tendermint

Reply:
Whats wrong with a module depending on tendermint? Ultimately setting validators is a tendermint feature. Creating a wrapper type doesn't remove the dependency for that module.